src/ast_parser/test.rs

```diff
@@ -379,3 +379,351 @@ fn fn_test() {
     }
     "###);
 }
+
+#[test]
+fn example_vec_temp() {
+    let p = expect_parse(
+        "
+        let x: i32;
+        let v: Vec<&'v mut i32>;
+        let p: &'p i32;
+        let tmp: &'tmp0 mut Vec<&'tmp1 mut i32>;
+
+        bb0: {
+            x = 22;
+            v = Vec_new();
+            p = &'L_x x;
+            tmp = &'L_v mut v;
+            Vec_push(move tmp, move p);
+            x = 44;
+            Vec_len(copy v);
+        }
+    ",
+    );
+    insta::assert_debug_snapshot!(p, @r###"
```

Comment:
My impression of `insta` was that it allowed you to avoid having large "expected" values in your source. Written this way, it's basically `assert`, no?

Reply:
indeed. I'm not sure whether these specific parser tests I added were of high value, but did help me while figuring out how the parser behaved, and I was thinking we could remove them altogether.

I can turn them into out-of-band snapshots, or delete them if we don't think they're that useful, wdyt ?

Reply:
I like out-of-band snapshots. I think it's nice to have a reference for what the AST looks like when writing visitors and such.

Reply:
I've now extracted these 2 tests into out-of-band snapshots, but kept the same Debug format for now (it turns out all the macros can work with inline or file snapshots equally as easily), let me know if you like it